Extensions/PathfindingBehavior/pathfindingruntimebehavior.ts

```diff
@@ -330,6 +407,27 @@ namespace gdjs {
       //Not path found
       this._pathFound = false;
     }
+    
+    static deepCloneHitboxes(hitboxes: Polygon[]): Polygon[] {
```

Comment:
Might be worth thinking about putting this as a static of `gdjs.Polygon` or `gdjs.RuntimeObject`.

Reply:
I put a deepClone method for one polygon in Polygon. I don't know if the deepClone of Polygon[] is useful and where it should go. It's almost a one-liner now.

Reply:
Looks good 👍

Reply:
>       * Do you think you could extract them + the RBush in a separate PR? That would be a much easier PR to review, and this one would become simpler (and safer if we add even more tests :))

I'll continue to write tests and make the PR after that because I would like to split the test and the RBush in 2 commits to make it easier to run the tests on the untouch implementation.